editor/plugins/texture_region_editor_plugin.cpp

```diff
@@ -510,8 +510,8 @@ void TextureRegionEditor::_scroll_changed(float) {
 }
 
 void TextureRegionEditor::_set_snap_mode(int p_mode) {
-	snap_mode_button->get_popup()->set_item_checked(snap_mode, false);
 	snap_mode = p_mode;
+	snap_mode_button->get_popup()->set_item_checked(snap_mode, false);
```

Comment:
Not familiar with this code so I might be wrong, but it seems like the expected behaviour here is to uncheck the previous snap mode *before* setting the new one. So the new code would be wrong (it would unset and then reset the same mode).

The uninitialized snap_mode should likely be set in the constructor.

Reply:
CC @groud

Reply:
I never worked on the texture region editor, bu yes, I don't think this should be moved.